### DISCUSSION TITLE
Whether same data could be inserted successfuly to Milvus or not

### DISCUSSION THREAD

**Discussion by gobigrassland** [milvus-io/milvus#17095] [Q&A and General discussion]
I inserted data (10x512, each 512-d vector is embedding of an image) into milvus,  and the numer of entities was 10.  I inserted the same data into miluvs, the number  was 20.  when I inserted the same data repeatedly， the numer of enties kept increasing.  Whether same data could be inserted successfuly to Milvus or not? But when I start query op with setting limt=100, only 10 samples was returned. 

the result of the code was:

num_entities: 10
num_entities: 20
num_entities: 30
num_entities: 40
num_entities: 50
num_entities: 60
num_entities: 70
num_entities: 80
num_entities: 90
num_entities: 100
num_entities: 110
num_entities: 120
num_entities: 130
num_entities: 140
num_entities: 150
num_entities: 160
num_entities: 170
num_entities: 180
num_entities: 190
num_entities: 200
0 [0, 8, 9, 3, 6, 5, 7, 1, 2, 4] [0.0, 1.7060574293136597, 1.7732272148132324, 1.8202385902404785, 1.8432246446609497, 1.860347867012024, 1.9747209548950195, 1.9785206317901611, 2.017852783203125, 2.0806241035461426]

-------------------------------------------------------------------------code -----------------------------------------------------------------------------

`def create_collection():

    face_idx = FieldSchema(
        name="image_idx",
        dtype=DataType.INT64,
        is_primary=True)

    face_feature = FieldSchema(
        name="image_feature",
        dtype=DataType.FLOAT_VECTOR,
        dim=512)

    schema = CollectionSchema(
        fields=[face_idx,face_feature],
        description="Test image search")

    collection_name = "image_search"

    collection = Collection(
        name=collection_name,
        schema=schema,
        shards_num=2,
        consistency_level="Strong")`

`def insert_data(data):

    collection = Collection("image_search")
    data_inserted = [[i for i in range(len(data))], data]
    collection.insert(data_inserted)
    print('num_entities:', collection.num_entities)`

`def create_index():

    collection = Collection("image_search")
    index_params = {
            "metric_type":"L2",
            "index_type":"IVF_FLAT",
            "params":{"nlist":1024}
        }

    collection.create_index(
        field_name="image_feature",
        index_params=index_params
        )
`

`def search(data_query, topk=10):

    collection = Collection("image_search")
    collection.load()
    search_params = {
        "metric_type": "L2",
        "params": {"nprobe": 10}
        }
    result = collection.search(
        data=data_query,
        anns_field="image_feature",
        param=search_params,
        limit=topk,
        expr=None,
        consistency_level="Strong"
        )

    for i in range(len(result)):
        print (i, result[i].ids, result[i].distances)
`

`if __name__ == '__main__':

    create_collection()
    data = np.load('feat.npy')
    n = 20
    for i in range(20):
        insert_data(data)
    create_index()
    search([data[0,:]], 100)
`

**Comment by yhmo** [comment]
The "image_idx" is primary key, you insert id=[0, 1, 2, 3, .... ,9] by 10 times. There are 200 entities in the collection, only 10 unique id, so the milvus reduce the result and only return 10 id-distance pairs.
